sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -356,6 +356,8 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     // scalastyle:on
     checkEvaluation(
       SubstringIndex(Literal("www||apache||org"), Literal( "||"), Literal(2)), "www||apache")
+    checkEvaluation(SubstringIndex(
+      Literal("www.apache.org"), Literal("."), Literal.create(null, IntegerType)), null)
```

Comment:
@miland-db For question 1, if we pass null as a parameter without `codegen`, it does not fail. Current master code with this test case passed(check logic of `checkEvaluation` will do `checkEvaluationWithoutCodegen`).

Reply:
@miland-db  For question 2, if we pass some other string value instead of null (no matter legal integer or not) with `codegen`, it fails. I will add another test case.